src/formatting.rs

```diff
@@ -710,45 +715,57 @@ fn parse_crate(
 struct SilentOnIgnoredFilesEmitter {
     ignore_path_set: Rc<IgnorePathSet>,
     source_map: Rc<SourceMap>,
-    emitter: EmitterWriter,
+    emitter: Box<dyn Emitter + Send>,
     has_non_ignorable_parser_errors: bool,
     can_reset: Rc<RefCell<bool>>,
 }
 
+impl SilentOnIgnoredFilesEmitter {
```

Comment:
The changes in e72f307f15138a8bb4fd1ae9ef3256fe1fa9dd94 on the emitters (and associated unit tests below) are a backport of a bug fix that's already been merged into master/2.x (https://github.com/rust-lang/rustfmt/pull/3920#pullrequestreview-317346826), but is missing from rustfmt 1.4.11.

This bug has been seen by a few users, including in the main rust-lang/rust repo (https://github.com/rust-lang/rustfmt/pull/3972#issuecomment-567566213) so I figured it'd be good to get this fix in along with the rustc-ap-* bump to get the bug fixed in rustfmt 1.x

Reply:
Thanks!